hail/python/hail/expr/functions.py

```diff
@@ -3384,9 +3384,45 @@ def zip(*arrays, fill_missing: bool = False) -> ArrayExpression:
                           aggregations)
 
 
+@typecheck(a=expr_array(), start=expr_int32, index_first=bool)
```

Comment:
does this need to be int32?

Reply:
Yes, because currently our arrays store their length as an int32, so that's what `hl.range` expects. All the ndarray stuff stores the shape as an int64 because I imagine we will eventually want to blow past 2^31 as an array length limit.